lib/ansible/modules/system/modprobe.py

```diff
@@ -97,11 +98,17 @@ def main():
             module_file = '/' + name + '.ko'
             builtin_path = os.path.join('/lib/modules/', uname_kernel_release.strip(),
                                         'modules.builtin')
-            with open(builtin_path) as builtins:
-                for line in builtins:
-                    if line.endswith(module_file):
-                        present = True
-                        break
+            try:
+                with open(builtin_path) as builtins:
+                    for line in builtins:
+                        if line.endswith(module_file):
+                            present = True
+                            break
+            except IOError as e:
+                # The open() failed
+                if e.errno != errno.ENOENT:
+                    # We are ok if this file doesnt exist, but nothing else
+                    module.fail_json(msg=to_native(e), exception=traceback.format_exc(), **result)
     except IOError as e:
         module.fail_json(msg=to_native(e), exception=traceback.format_exc(), **result)
```

Comment:
What is the value of failing if   /proc/modules or modules.builtin is unreadable?  Maybe replace with warn if you want to tell the user about builtin_path.  Later, modprobe will return an error if it is builtin.

Reply:
I was trying to be specific and only fail when `errno.ENOENT` occurs (file doesn't exist to read from), but if it is not readable by the user that is a different issue that I don't have a good test case to validate with.

It appears that the Red Hat EL 6 "2.6.32-754.15.3.el6.x86_64" kernel package deletes this when the kernel is upgraded (I need to do more testing), but missing either of these is still apparently a valid (albeit non-optimal) configuration.

Reply:
I don't have a good reason why the file would exist but not readable, they are readable in packaged kernels. What I am suggesting is to move this exception in reading the builtins file away from failing the module, as it has proven a little bit tricky. I consider modprobe return code the authority on whether it failed or not.